addons/web_editor/static/src/js/editor/snippets.options.js

```diff
@@ -2213,9 +2219,13 @@ const ListUserValueWidget = UserValueWidget.extend({
         this.containerEl.appendChild(tableWrapper);
         this.el.classList.add('o_we_fw');
         this._makeListItemsSortable();
+        this.isVisibilityCondition = this.options.dataAttributes.attributeName == 'visibilityCondition';
         if (this.createWidget) {
             return this.createWidget.appendTo(this.containerEl);
         }
+        if (this.isVisibilityCondition) {
```

Comment:
:fearful: It's scary to do changes here in this file. as far as i see, you used `_populateDependencyOptions()` just to render the option value in list which should be done around `_renderCustomXML` method in options.js. `ListUserValueWidget` is common widget used at many place, we can't do snippet specific code here. 

I prefer not to touch this widget as possible as, just refer how we populated xml for the `equal to`  & `!equal to`.

Reply:
Yes, I understand your concern about modifying the `ListUserValueWidget` since it's a common widget used in multiple places. The approach of handling this in `_renderCustomXML` similar to how "equal to" and "not equal to" is handled for `we-select` widget.   I faced a specific challenge with this approach:

   - First, we populate `<we-list>` XML with `we-button` elements in `_renderCustomXML` widget. 
   - Then, `_renderXMLWidgets` is called to register widgets and sub-widgets
   - When the `we-list` widget's `start` method executes, it replaces the content with an empty `table` element
   - As a result, the `we-button` widgets never get registered as sub-widgets of `we-list` and the `dataSelectAttribute` method will also not be present in `this._methodsNames` because  `this._userValueWidget` array is empty for `we-list` widget. 
   
  `_populateDependencyOptions` method is responsible for fetching the related dom element and collect the necessary information about that element so that we can pass that info into `_addItemToTable` method. We cannot remove that method from `ListUserValueWidget` since it is necessary .